server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -184,6 +186,70 @@ def dict(self, *, exclude_none: bool = True, **kwargs) -> dict:
 ]
 
 
+def remove_void_conditions_from_filter_steps(
+    steps: list[PipelineStepWithVariables | PipelineStep | BaseStep],
+) -> list[PipelineStepWithVariables | PipelineStep | BaseStep]:
+    """
+    This method will remove all FilterStep with conditions having "__VOID__"
+    in them. either the "value" key or the "column" key.
+    """
+
+    def _contains_void_as_value(value: dict) -> bool:
+        return any(
+            [
+                value.get("value") == "__VOID__",
+                value.get("column") == "__VOID__",
+            ]
+        )
+
+    def _remove_void_value_elements(obj: dict[str, Any] | list[dict[str, Any]]) -> None:
+        if isinstance(obj, dict):
+            keys_to_remove = [
+                k for k, v in obj.items() if isinstance(v, dict) and _contains_void_as_value(v)
+            ]
+            for key in keys_to_remove:
+                del obj[key]  # remove condition if void is found
+
+            for v in obj.values():
+                _remove_void_value_elements(v)
+
+            if "condition" in obj:
+                if ("and_" in obj["condition"] and obj["condition"]["and_"] == []) or (
+                    "or_" in obj["condition"] and obj["condition"]["or_"] == []
+                ):
+                    del obj["condition"]
+
+        elif isinstance(obj, list):
+            indices_to_remove = [
+                i for i, v in enumerate(obj) if isinstance(v, dict) and _contains_void_as_value(v)
+            ]
+
+            for index in sorted(indices_to_remove, reverse=True):
+                del obj[index]
+            for v in obj:
+                _remove_void_value_elements(v)
+
+            for index, v in enumerate(obj):
+                if "or_" in v and v["or_"] == []:
+                    del obj[index]
+                if "and_" in v and v["and_"] == []:
+                    del obj[index]
```

Comment:
I don't think these need to be closures  ?

Reply:
you're right, updated !